src/plugins/yy/yyplugin.cpp

```diff
@@ -52,6 +52,33 @@ static QString sanitizeName(QString name)
     return name.replace(regexp, QStringLiteral("_"));
 }
 
+/**
+ * Determine the resource identifier associated with the given filepath.
+ * This is achieved by looking for a *.yy meta file and, if found, returning
+ * its name without the extension. The function starts searching the current
+ * directory and up to two parent directories to account for a typical
+ * project directory structure.
+ * If no *.yy file can be found, returns the filename without its extension.
+ */
+static QString determineResourceId(const QString &filePath)
+{
+    constexpr unsigned int searchDepth = 3;
+    QFileInfo fi(filePath);
+    QDir dir(fi.canonicalPath());
+    dir.setNameFilters(QStringList("*.yy"));
+
+    for (unsigned int i = 0; i < searchDepth; ++i) {
+        if (i > 0 && !dir.cdUp())
+            break;
+
+        const QFileInfoList &files = dir.entryInfoList();
```

Comment:
Since we're only interested in the first file, I think we could do:
```c++
const QString yyFile = QDirIterator(dir).next();
if (!yyFile.isEmpty())
    return QFileInfo(yyFile).completeBaseName();
```

Reply:
Done.